pkg/ebpf/processor_proctree.go

```diff
@@ -0,0 +1,245 @@
+
+	// Up Parent (Up in hierarchy until parent is a process and not a lwp)
+	parentTid, err := parse.ArgVal[int32](event.Args, "up_parent_tid")
+	errs = append(errs, err)
+	parentNsTid, err := parse.ArgVal[int32](event.Args, "up_parent_ns_tid")
+	errs = append(errs, err)
+	parentPid, err := parse.ArgVal[int32](event.Args, "up_parent_pid")
+	errs = append(errs, err)
+	parentNsPid, err := parse.ArgVal[int32](event.Args, "up_parent_ns_pid")
+	errs = append(errs, err)
+	parentStartTime, err := parse.ArgVal[uint64](event.Args, "up_parent_start_time")
+	errs = append(errs, err)
+
+	// Thread Group Leader (might be the same as the "child", if "child" is a process)
+	leaderTid, err := parse.ArgVal[int32](event.Args, "leader_tid")
+	errs = append(errs, err)
+	leaderNsTid, err := parse.ArgVal[int32](event.Args, "leader_ns_tid")
+	errs = append(errs, err)
+	leaderPid, err := parse.ArgVal[int32](event.Args, "leader_pid")
+	errs = append(errs, err)
+	leaderNsPid, err := parse.ArgVal[int32](event.Args, "leader_ns_pid")
+	errs = append(errs, err)
+	leaderStartTime, err := parse.ArgVal[uint64](event.Args, "leader_start_time")
+	errs = append(errs, err)
+
+	// Child (might be a process or a thread)
+	childTid, err := parse.ArgVal[int32](event.Args, "child_tid")
+	errs = append(errs, err)
+	childNsTid, err := parse.ArgVal[int32](event.Args, "child_ns_tid")
+	errs = append(errs, err)
+	childPid, err := parse.ArgVal[int32](event.Args, "child_pid")
+	errs = append(errs, err)
+	childNsPid, err := parse.ArgVal[int32](event.Args, "child_ns_pid")
+	errs = append(errs, err)
+	childStartTime, err := parse.ArgVal[uint64](event.Args, "start_time") // child_start_time
+	errs = append(errs, err)
+
+	// Deal with errors
+	for _, err := range errs {
+		if err != nil {
+			return err
+		}
+	}
+
+	// Calculate hashes
+	childHash := utils.HashTaskID(uint32(childTid), uint64(childStartTime))
+	parentHash := utils.HashTaskID(uint32(parentTid), uint64(parentStartTime))
+	leaderHash := utils.HashTaskID(uint32(leaderTid), uint64(leaderStartTime))
+
+	return t.processTree.FeedFromFork(
+		proctree.ForkFeed{
+			TimeStamp:       childStartTime, // event timestamp is the same
+			ChildHash:       childHash,
+			ParentHash:      parentHash,
+			LeaderHash:      leaderHash,
+			ParentTid:       parentTid,
+			ParentNsTid:     parentNsTid,
+			ParentPid:       parentPid,
+			ParentNsPid:     parentNsPid,
+			ParentStartTime: parentStartTime,
+			LeaderTid:       leaderTid,
+			LeaderNsTid:     leaderNsTid,
+			LeaderPid:       leaderPid,
+			LeaderNsPid:     leaderNsPid,
+			LeaderStartTime: leaderStartTime,
+			ChildTid:        childTid,
+			ChildNsTid:      childNsTid,
+			ChildPid:        childPid,
+			ChildNsPid:      childNsPid,
+			ChildStartTime:  childStartTime,
+		},
+	)
+}
+
+// procTreeForkRemoveArgs removes arguments needed for the process tree only (when source is events).
+func (t *Tracee) procTreeForkRemoveArgs(event *trace.Event) error {
```

Comment:
I don't like the idea here of inconsistent event through the pipeline, even if it only adding arguments.
I feel we might have troubles with it in the future.

Reply:
I thought it was creative :P. My problem with another event is to have only for SchedProcessFork and not for the others, seems weird as well.

Reply:
https://github.com/aquasecurity/tracee/pull/3498#discussion_r1340157044